clinica/iotools/utils/data_handling.py

```diff
@@ -639,3 +655,298 @@ def center_all_nifti(bids_dir, output_dir):
+            # - aligned to the left, with the size of the column
+            # s = string, f = float
+            # . for precision with float
+            # https://docs.python.org/2/library/stdtypes.html#string-formatting for more information
+            warning_message += ('%-' + str(file_width) + 's%-' + str(center_width) + 's%-25.2f\n') % (basename(file),
+                                                                                                      str(center),
+                                                                                                      l2)
+
+        warning_message += '\nClinica provides a tool to counter this problem by replacing the center of the volume' \
+                           + ' at the origin of the world coordinates.\nUse the following command line to correct the '\
+                           + 'header of the faulty NIFTI volumes in a new folder:\n' + Fore.RESET \
+                           + Fore.BLUE + '\nclinica iotools center-nifti ' + abspath(bids_dir) + ' ' \
+                           + abspath(bids_dir) + '_centered\n\n' + Fore.YELLOW \
+                           + 'You will find more information on the command by typing ' + Fore.BLUE \
+                           + 'clinica iotools center-nifti' + Fore.YELLOW + ' in the console.\nDo you still want to '\
+                           + 'launch the pipeline now ?' + Fore.RESET
+        cprint(warning_message)
+        while True:
+            cprint('Your answer [yes/no]:')
+            answer = input()
+            if answer.lower() in ['yes', 'no']:
+                break
+            else:
+                cprint(Fore.RED + 'You must answer yes or no' + Fore.RESET)
+        if answer.lower() == 'no':
+            cprint(Fore.RED + 'Clinica will now exit...' + Fore.RESET)
+            sys.exit(0)
+
+
+def is_centered(nii_volume, threshold_l2=80):
+    """
+    Tells if a NIfTI volume is centered on the origin of the world coordinate system.
+
+    SPM has troubles to segment files if the center of the volume is not close from the origin of the world coordinate
+    system. A series of experiment have been conducted: we take a volume whose center is on the origin of the world
+    coordinate system. We add an offset using coordinates of affine matrix [0, 3], [1, 3], [2, 3] (or by modifying the
+    header['srow_x'][3], header['srow_y'][3], header['srow_z'][3], this is strictly equivalent).
+
+    It has been determined that volumes were still segmented with SPM when the L2 distance between origin and center of
+    the volume did not exceed 100 mm. Above this distance, either the volume is either not segmented (SPM error), or the
+    produced segmentation is wrong (not the shape of a brain anymore)
+
+    Args:
+        nii_volume: path to NIfTI volume
+        threshold_l2: maximum distance between origin of the world coordinate system and the center of the volume to
+                    be considered centered
```

Comment:
Write here that 80 was determined empirically after several trials on <XX> dataset(s).

Reply:
Done